service/storage_proxy.cc

```diff
@@ -2153,6 +2161,12 @@ storage_proxy::mutate_atomically(std::vector<mutation> mutations, db::consistenc
         };
 
         future<> run() {
+            if (!std::exchange(_checked_cdc, true) && _p.get_cdc_service()->needs_cdc_augmentation(_mutations)) {
+                return _p.get_cdc_service()->augment_mutation_call(_timeout, std::move(_mutations)).then([this](std::tuple<std::vector<mutation>, cdc::result_callback>&& t) {
+                    _mutations = std::move(std::get<0>(t));
+                    return run();
+                });
+            }
```

Comment:
Why not call augment code before creating a context?

Reply:
It was just easier to put the future-stuff inside the context. No other reason.

Reply:
But them you need to do tricks with a bool flag.

Reply:
This is true. Now I however remember. There is actually a much better reason why it is inside the context object: Post-image. 
The cdc API here gives optionally back a post-image-generation object that iff not empty need to be called after the mutation is done. This is not yet actually used, _but_ when writing so that it is, it will be easier if it exists as a result of the above call inside the context. Alternative is to make it a field in the context iff augmentation returned it.

Reply:
What do you mean by "after the mutation is done"?

Reply:
I mean after "run()" has successfully finished. When actual handling of post-image is implemented, the idea is that the code here would be something like:
```
auto post_handler = std::get<1>(t);
auto f = run();
if (handler) { 
  f = f.then(post_handler);
}
```
Now, arguably this could be handled just as well outside, so I guess it is not a super strong argument. I however still maintain that having the code outside the context object gets more messy, since it adds a conditional continuation _before_ creating the context object, which means at least some duplicated code.

Reply:
We are entering personal preferences territory here. IMO the code that do something, set flag to not do it next time and than runs itself recursively may look nice what you write it since it saves typing, but after a year, when you'll forget all about it, will be harder to understand. Code duplication may be less nice, but you did it in `storage_proxy::mutate` (just because you had no place to store a flag :))

Reply:
Ok. Updated to do the deed outside.